internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -168,12 +182,30 @@ func (f *fleetGateway) worker() {
 				continue
 			}
 
-			actions := make([]fleetapi.Action, len(resp.Actions))
-			for idx, a := range resp.Actions {
-				actions[idx] = a
+			actions := f.queueScheduledActions(resp.Actions)
+			actions, err = f.dispatchCancelActions(actions)
+			if err != nil {
+				f.log.Error(err.Error())
+				// TODO set errMsg? update status?
 			}
 
+			queued, expired := f.gatherQueuedActions(ts)
+			f.log.Debugf("Gathered %d actions from queue, %d actions expired", len(queued), len(expired))
+
+			// TODO update all actions in expired as aborted?
```

Comment:
Do we need to send any acks for actions we won't run?

Reply:
We probably will not use these acks in Kibana

Reply:
I don't think we need to ack on expired action, I believe it would be part of the UI to display where is the update gab wdyt @joshdover  ?